common/stats/stats.go

```diff
@@ -0,0 +1,479 @@
+// This package borrows from go-metrics so we can avoid exposing dependencies,
+// and to give callers more flexibility in recording and formatting scoped stats.
```

Comment:
carealot had 2k+ loc to mimic twitter stats, this is about 1/4 of that and basically a full rewrite.

We still use go-metrics for the default registry and instruments, but that isn't exposed to the user. User can go with the defaults or override the registry impl and/or individual instrument impls.